src/RoslynPad.Common.UI/ViewModels/OpenDocumentViewModel.cs

```diff
@@ -85,7 +91,29 @@ private set
             }
         }
 
-        public DocumentViewModel Document { get; private set; }
+        public DocumentViewModel Document
+        {
+            get => _document;
+            private set
+            {
+                var oldDocument = _document;
+                if (!SetProperty (ref _document, value))
+                    return;
+
+                if (oldDocument != null)
+                    oldDocument.PropertyChanged -= DocumentOnPropertyChanged;
+                if(_document != null)
+                    _document.PropertyChanged += DocumentOnPropertyChanged;
+            }
+        }
+
+        private void DocumentOnPropertyChanged (object sender, PropertyChangedEventArgs args)
```

Comment:
This is not a good editing experience IMO. If someone changes a file of a document that's currently being edited, when saving, I'd expect the original name and path to hold (that's what Notepad++ does, for example). Perhaps when opening a document, we should clone the `DocumentViewModel` and keep the clone disconnected from the watcher completely. This way the tree can update freely. 

I think you can pretty much undo all changes to this file, and add the cloning at `SetDocument`.